jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamInitRunner.kt

```diff
@@ -0,0 +1,51 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda.execution.sam
+
+import com.intellij.execution.configurations.GeneralCommandLine
+import com.intellij.execution.process.CapturingProcessHandler
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.openapi.vfs.VfsUtil
+import com.intellij.openapi.vfs.VirtualFile
+import software.amazon.awssdk.services.lambda.model.Runtime
+import software.aws.toolkits.jetbrains.settings.SamSettings
+
+class SamInitRunner {
+    private val samCliExecutable = SamSettings.getInstance().executablePath
+    private var commandLine = GeneralCommandLine(samCliExecutable)
+            .withParentEnvironmentType(GeneralCommandLine.ParentEnvironmentType.CONSOLE)
+            .withParameters("init")
+            .withParameters("--no-input")
+    private lateinit var outputDir: VirtualFile
+    fun applyLocation(location: String) = applyParameter("--location", location)
+
+    fun applyName(name: String) = applyParameter("--name", name)
+
+    fun applyOutputDir(location: VirtualFile) = apply {
+        outputDir = location
+    }
+
+    fun applyRuntime(runtime: Runtime) = applyParameter("--runtime", runtime.toString())
+
+    private fun applyParameter(flag: String, value: String) = apply {
+        commandLine = commandLine.withParameters(flag)
+            .withParameters(value)
+    }
+
+    fun execute() = ApplicationManager.getApplication().runWriteAction {
+        // set output to a temp dir
+        val tempDir = LocalFileSystem.getInstance().findFileByIoFile(createTempDir())
+        applyParameter("--output-dir", tempDir!!.path)
+
+        // run
+        val process = CapturingProcessHandler(commandLine).runProcess()
+        if (process.exitCode != 0) {
+            throw RuntimeException("Could not execute `sam init`!: ${process.stderrLines.last()}")
```

Comment:
Not sure if this fits into this bucket, but any exception that might be surfaced to the user should pull the message from a `message` resource to be able to support internationalization in the future.

Reply:
done